test/integration/apiserver/cel/validatingadmissionpolicy_test.go

```diff
@@ -0,0 +1,937 @@
+	"context"
+	"strings"
+	"testing"
+	"time"
+
+	genericfeatures "k8s.io/apiserver/pkg/features"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	featuregatetesting "k8s.io/component-base/featuregate/testing"
+	apiservertesting "k8s.io/kubernetes/cmd/kube-apiserver/app/testing"
+	"k8s.io/kubernetes/test/integration/framework"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+
+	v1 "k8s.io/api/core/v1"
+
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	admissionregistrationv1alpha1 "k8s.io/api/admissionregistration/v1alpha1"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+// Test_ValidateNamespace_NoParams tests a ValidatingAdmissionPolicy that validates creation of a Namespace with no params.
+func Test_ValidateNamespace_NoParams(t *testing.T) {
+	failurePolicy := admissionregistrationv1alpha1.Fail
+	ignorePolicy := admissionregistrationv1alpha1.Ignore
+
+	testcases := []struct {
+		name          string
+		policy        *admissionregistrationv1alpha1.ValidatingAdmissionPolicy
+		policyBinding *admissionregistrationv1alpha1.ValidatingAdmissionPolicyBinding
+		namespace     *v1.Namespace
+		err           string
+		failureReason metav1.StatusReason
+	}{
+		{
+			name: "namespace name contains suffix enforced by validating admission policy, using object metadata fields",
+			policy: &admissionregistrationv1alpha1.ValidatingAdmissionPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "validate-namespace-suffix",
+				},
+				Spec: admissionregistrationv1alpha1.ValidatingAdmissionPolicySpec{
+					MatchConstraints: &admissionregistrationv1alpha1.MatchResources{
+						ResourceRules: []admissionregistrationv1alpha1.NamedRuleWithOperations{
+							{
+								RuleWithOperations: admissionregistrationv1alpha1.RuleWithOperations{
+									Operations: []admissionregistrationv1.OperationType{
+										"CREATE",
+									},
+									Rule: admissionregistrationv1.Rule{
+										APIGroups: []string{
+											"",
+										},
+										APIVersions: []string{
+											"*",
+										},
+										Resources: []string{
+											"namespaces",
+										},
+									},
+								},
+							},
+						},
+					},
+					Validations: []admissionregistrationv1alpha1.Validation{
+						{
+							Expression: "object.metadata.name.endsWith('k8s')",
+						},
+					},
+					FailurePolicy: &failurePolicy,
+				},
+			},
+			policyBinding: &admissionregistrationv1alpha1.ValidatingAdmissionPolicyBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "validate-namespace-suffix-binding",
+				},
+				Spec: admissionregistrationv1alpha1.ValidatingAdmissionPolicyBindingSpec{
+					PolicyName: "validate-namespace-suffix",
+				},
+			},
+			namespace: &v1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-k8s",
+				},
+			},
+			err: "",
+		},
+		{
+			name: "namespace name does NOT contain suffix enforced by validating admission policyusing, object metadata fields",
```

Comment:
Let's add a version of this test, but using Forbidden as the expected status reason

Reply:
Fixed!